admin/admin-service/src/main/java/io/mosip/admin/service/impl/ApplicantDetailServiceImpl.java

```diff
@@ -0,0 +1,144 @@
+import io.mosip.admin.packetstatusupdater.util.AuditUtil;
+import io.mosip.admin.packetstatusupdater.util.EventEnum;
+import io.mosip.admin.service.ApplicantDetailService;
+import io.mosip.admin.util.CbeffToBiometricUtil;
+import io.mosip.admin.util.RestClient;
+import io.mosip.admin.util.Utility;
+import io.mosip.biometrics.util.ConvertRequestDto;
+import io.mosip.biometrics.util.face.FaceDecoder;
+import io.mosip.kernel.core.idvalidator.exception.InvalidIDException;
+import io.mosip.kernel.core.idvalidator.spi.RidValidator;
+import io.mosip.kernel.core.util.JsonUtils;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.codec.binary.StringUtils;
+import org.json.JSONException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.ResourceAccessException;
+
+import java.util.*;
+
+@Service
+public class ApplicantDetailServiceImpl implements ApplicantDetailService {
+
+    private static final String FACE = "Face";
+
+    @Autowired
+    RestClient restClient;
+
+    @Autowired
+    private RidValidator<String> ridValidator;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AuditUtil auditUtil;
+
+    @Autowired
+    CbeffToBiometricUtil cbeffToBiometricUtil;
+
+
+    @Autowired
+    Utility utility;
+
+    private static final String IDENTITY = "identity";
+
+    private static final String RESPONSE = "response";
+
+    private static final String DOCUMENTS="documents";
+
+    private static final String ApplicantPhoto = "applicantPhoto";
+
+    private static final String VALUE = "value";
+    private static final String DOB = "dob";
+
+    @Value("${mosip.admin.applicantDetails}")
+    private String[] applicantDetails;
+
+
+    @Override
+    public ApplicantDetailsDto getApplicantDetails(String rid) throws Exception {
+        ApplicantDetailsDto applicantDetailsDto =new ApplicantDetailsDto();
+        Map<String,String> applicantDataMap=new HashMap<>();
+        ConvertRequestDto convertRequestDto = new ConvertRequestDto();
+        List<String> pathsegments=new ArrayList<>();
+        String dateOfBirth=null;
+        String individualBiometrics=null;
+        pathsegments.add(rid);
+        String imageData=null;
+        try {
+			if (!ridValidator.validateId(rid)) {
+				throw new RequestException(ApplicantDetailErrorCode.RID_INVALID.getErrorCode(),
+                        ApplicantDetailErrorCode.RID_INVALID.getErrorMessage());
+			}
+            String response = restClient.getApi(ApiName.RETRIEVE_IDENTITY_API,pathsegments,"type","bio",String.class);
+            JSONObject responseObj= objectMapper.readValue(response,JSONObject.class);
+            if(response!=null && responseObj.containsKey("response")) {
```

Comment:
Please inverse this condition and handle else condition first to avoid below nested loops

Reply:
taken care

Reply:
Since if is throwing a exception, you don't need else

Reply:
Also as per the new condition what if response was null

Reply:
then will throw error The card for this request ID is not generated. Please check the status of the ID.